optimus/io/driver.py

```diff
@@ -1,31 +1,332 @@
+from __future__ import annotations
+
+from _ast import Tuple
+from abc import abstractmethod, ABC
 from enum import Enum, unique
 
 
+class DriverContext:
+    def __init__(self, strategy: DriverStrategy) -> None:
+        self.strategy = strategy
+
+    @property
+    def strategy(self) -> DriverStrategy:
+        return self.strategy
+
+    @strategy.setter
+    def strategy(self, strategy: DriverStrategy) -> None:
+        self.strategy = strategy
+
+    def get_driver_properties(self) -> DriverProperties:
+        return self.strategy.get_driver_properties()
+
+    def get_url(self, *args) -> str:
+        return self.strategy.get_url(args)
+
+    def get_query_table_names(self, *args) -> str:
+        return self.strategy.get_query_table_names(args)
+
+    def get_query_table_name(self, *args) -> str:
+        return self.strategy.get_query_table_name(args)
+
+    def get_count_query(self, db_table) -> str:
+        return self.strategy.get_count_query(db_table)
+
+
+class DriverStrategy(ABC):
```

Comment:
Not sure if "DriverStrategy" can help other devs to know easily what is this class about. Maybe something like AbstractDriver?

Reply:
What you think about `AbstractJdbcDriverStrategy`?